### PR TITLE
[CCTRI-2252] - Daily requests for Event API

### PR DESCRIPTION
**Pull request by ytsek**
7 daily requests for Event API instead of one 7-day request

### CODE REVIEW

File: code/api/integration.py [hunk 1]
```diff
@@ -111,22 +112,33 @@ def get_events_for_detection(key, detection_uuid):
...
-
-    if error:
-        return None, error
-
-    events = data['events']
+    events = []
+    now = datetime.datetime.now()
+    end_date = now.isoformat()
+    start_date = (now - datetime.timedelta(days=1)).isoformat()
+    day_range = 7
```
Comment (AnastasiiaRozlyvan) [inline code/api/integration.py:127]: I think this constant should be moved to the config file.

File: code/api/integration.py [hunk 2]
```diff
@@ -111,22 +112,33 @@ def get_events_for_detection(key, detection_uuid):
...
+    while day_range and len(events) < limit:
+        json = {
+            'query': f"{observable['type']} = '{observable['value']}'",
+            'start_date' : mil_time(start_date),
```
Comment (AnastasiiaRozlyvan) [inline code/api/integration.py:131]: Please run `flake8`. I think there may be some `warnings` here that should be fixed.

File: code/api/integration.py (hunk 1 above)
Comment (ytsek) [reply code/api/integration.py]: I have made some improvements and add constant variable.

File: code/api/integration.py (hunk 2 above)
Comment (ytsek) [reply code/api/integration.py]: Already fixed.

File: code/api/integration.py [hunk 3]
```diff
@@ -111,22 +112,34 @@ def get_events_for_detection(key, detection_uuid):
...
+            'end_date': mil_time(end_date)
+        }
+        data, error = _request('POST', url, key=key, json=json)
+        if error:
+            return None, error
+        end_date, start_date = start_date, (
+            datetime.datetime.fromisoformat(start_date) -
+            datetime.timedelta(days=1)
+        ).isoformat()
+        events.extend(data['events'])
```
[the remark this answers is not in the document]
Comment (ytsek) [reply code/api/integration.py:141]: Have implemented events limit logics.

**Review by oshynk — APPROVED** [review]

File: code/tests/unit/api/test_integration.py [hunk 4]
```diff
@@ -274,17 +278,20 @@ def test_get_events_success(client, gti_api_request):
...
         expected_method,
         expected_url,
         headers=expected_headers,
         json=expected_json,
     )
 
-    assert events == expected_events
+    gti_api_request.call_count == 7
+
+    assert events[:10] == expected_events
```
Comment (AnastasiiaRozlyvan) [inline code/tests/unit/api/test_integration.py:294]: As I understand it, now the function is called 7 times with the same parameters. Maybe it is better not to use `.return_value` in this case but use `.side_effect` and for example, return 10 events once, and then return an empty dictionary? Then you will not need to do a slice of the first 10 events.

File: code/tests/unit/api/test_integration.py [hunk 5]
```diff
@@ -274,17 +278,20 @@ def test_get_events_success(client, gti_api_request):
...
-    gti_api_request.assert_called_once_with(
+    gti_api_request.assert_any_call(
         expected_method,
         expected_url,
         headers=expected_headers,
         json=expected_json,
     )
 
-    assert events == expected_events
+    gti_api_request.call_count == 7
```
Comment (AnastasiiaRozlyvan) [inline code/tests/unit/api/test_integration.py:292]: Maybe 
```python
assert gti_api_request.call_count == 7
```

**Review by AnastasiiaRozlyvan — APPROVED** [review]

**Pull request merged**